airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/DbSourceDiscoverUtil.java

```diff
@@ -52,11 +53,19 @@ public static <DataType> void logSourceSchemaChange(final Map<String, TableInfo<
           .distinct()
           .collect(Collectors.toList());
       final JsonNode currentJsonSchema = fieldsToJsonSchema(fields);
-
+      final List<String> mismatchedFields = new ArrayList<>();
       final JsonNode catalogSchema = stream.getJsonSchema();
-      if (!catalogSchema.equals(currentJsonSchema)) {
+      catalogSchema.get("properties").fieldNames().forEachRemaining(fieldName -> {
+        if (!currentJsonSchema.get("properties").has(fieldName) ||
+            !currentJsonSchema.get("properties").get(fieldName).equals(catalogSchema.get(fieldName))) {
```

Comment:
isn't `catalogSchema.get(fieldName)` the same as `fieldName` or did you mean to compare with the latter here?

Reply:
Possible to write a unit test for this?

Reply:
I wanted to compare the k:v pair within the `properties` key. e.g:

_Catalog:_
```
{
   "type":"object",
   "properties":{
      "updated_at":{
         "type":"string",
         "airbyte_type":"timestamp_without_timezone",
         "format":"date-time"
      },
      "id":{
         "type":"number",
         "airbyte_type":"integer"
      }
   }
}
```
_currentJson:_
```
{
   "type":"object",
   "properties":{
      "id":{
         "type":"number",
         "airbyte_type":"integer"
      },
      "updated_at":{
         "type":"string",
         "airbyte_type":"timestamp_without_timezone",
         "format":"date-time"
      },
      "_ab_cdc_lsn":{
         "type":"number"
      },
      "_ab_cdc_deleted_at":{
         "type":"string"
      },
      "_ab_cdc_updated_at":{
         "type":"string"
      }
   }
}
```